react/leanShipping.js

```diff
@@ -0,0 +1,379 @@
+import estimateCalculator from '@vtex/estimate-calculator'
+import sumBy from 'lodash/sumBy'
+import minBy from 'lodash/minBy'
+import isEqual from 'lodash/isEqual'
+import omit from 'lodash/omit'
+import { getStructuredOption } from './DeliveryPackagesUtils'
+import { CHEAPEST, COMBINED, FASTEST, DELIVERY } from './constants'
+import {
+  hasDeliveryWindows,
+  hasOnlyScheduledDelivery,
+  isPickup,
+  isDelivery,
+  isCurrentChannel,
+} from './utils'
+
+function getShippingEstimateInSeconds(shippingEstimate) {
+  return estimateCalculator.getShippingEstimateQuantityInSeconds(
+    shippingEstimate
+  )
+}
+
+function filterPreviousExpensiveOptions(options) {
+  return options.filter((option, index) =>
+    options[index + 1] ? option.price < options[index + 1].price : true
```

Comment:
What happens to the last one on the list?

Reply:
Nullpointer exception, array out of bounds